### PR TITLE
Use Koin to inject AsessmentRegistryProvider into AssessmentActivity

### PR DESCRIPTION
**Pull request by nategbrown9**

### CODE REVIEW

**Review by liujoshua — APPROVED** [review]

File: assessmentModel/src/commonMain/kotlin/serialization/ModuleInfoObject.kt [hunk 1]
```diff
@@ -21,12 +21,12 @@ val moduleInfoSerializersModule = SerializersModule {
 data class ModuleInfoObject(
     override val assessments: List<TransformableAssessment>,
     override var packageName: String? = null,
-    override val bundleIdentifier: String? = null): ResourceInfo, EmbeddedJsonModuleInfo {
+    override val bundleIdentifier: String? = null,
+    @Transient
+    override val jsonCoder: Json = Serialization.JsonCoder.default): ResourceInfo, EmbeddedJsonModuleInfo {
```
Comment (syoung-smallwisdom) [inline assessmentModel/src/commonMain/kotlin/serialization/ModuleInfoObject.kt:26]: So... the idea was that the module info objects could be deserialized with a "type" that is registered with the `moduleInfoSerializersModule` where each module would define it's own type. If you wanted to instantiate this thing directly and *not* use the default coder, then would probably be better to be explicit and have a *different* class such as:

```
class ModuleInfoImpl(
     override val assessments: List<TransformableAssessment>,
     override var resourceInfo: ResourceInfo,
     override val jsonCoder: Json
) : EmbeddedJsonModuleInfo 
```

**Review by syoung-smallwisdom — CHANGES_REQUESTED** [review]
Kinda want to keep the json coder as immutable for the type-based serialization of modules.

File: assessmentModel/src/commonMain/kotlin/serialization/ModuleInfoObject.kt (hunk 1 above)
Comment (nategbrown9) [reply assessmentModel/src/commonMain/kotlin/serialization/ModuleInfoObject.kt]: I went ahead and changed the sample app to load modules from a JSON file.

**Review by syoung-smallwisdom — APPROVED** [review]
👍

**Pull request merged**